src/configuration/preprocessors/api/index.coffee

```diff
@@ -1,32 +1,36 @@
 {async} = require "fairmont"
-extractParamters = require "./parameters"
+extractParameters = require "./parameters"
 extractCFr = require "./cfr"
 extractResources = require "./resources"
 extractActions = require "./actions"
 addResponses = require "./responses"
 selectRuntime = require "./runtime"
 addVariables = require "./variables"
 
-module.exports = async (description) ->
+module.exports = async (mungedConfig) ->
+  {name, env} = mungedConfig
+  mungedConfig.gatewayName = "#{name}-#{env}"
+  mungedConfig.policyName = "#{name}-#{env}"
+
   # Extract path and querystring parameter configuration
-  description = extractParamters description
+  mungedConfig = extractParameters mungedConfig
+
+  # Add environment varialbles that are injected into every Lambda.
+  mungedConfig = yield addVariables mungedConfig
 
   # Extract CloudFront configuration
-  description = yield extractCFr description
+  mungedConfig = yield extractCFr mungedConfig
 
   # Build up resource array that includes virtual resources needed by Gateway.
-  description = yield extractResources description
+  mungedConfig = yield extractResources mungedConfig
 
   # Compute the formatted template names for API action defintions.
-  description = yield extractActions description
+  mungedConfig = yield extractActions mungedConfig
 
   # Add the possible HTTP responses to every API action specification.
-  description = yield addResponses description
+  mungedConfig = yield addResponses mungedConfig
```

Comment:
this repeated modification is clunky. might be a good place for `go`:

`go (extractParameters description),  addVariables, extractCFr, extractResources, extractActions, addResponses`

#nit